components/selection/selection-summary.js

```diff
@@ -35,25 +36,54 @@ class Summary extends LocalizeCoreElement(SelectionObserverMixin(LitElement)) {
 	}
 
 	render() {
+		if (!this._summary) {
+			return nothing;
+		}
+		return html`
+			<p class="d2l-body-compact">${this._summary}</p>
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('_provider') || changedProperties.has('selectionInfo')) {
+			this._updateSelectSummary();
+		}
+	}
+
+	_updateSelectSummary() {
 		if (this._provider && this._provider.selectionSingle) return;
 
 		let count;
-		let summary;
 		if (this._provider && this._provider.selectionCountOverride !== undefined) {
 			count = this._provider.selectionCountOverride;
-			summary = (this._provider.selectionCountOverride === 0 && this.noSelectionText ?
-				this.noSelectionText : this.localize('components.selection.selected', 'count', count));
+			this._summary = this._provider.selectionCountOverride === 0 && this.noSelectionText ?
+				this.noSelectionText : this.localize('components.selection.selected', 'count', count);
 		} else {
-			count = (this.selectionInfo.state === SelectionInfo.states.allPages ?
-				this._provider.itemCount : this.selectionInfo.keys.length);
+			let includePlus = false;
+			if (this._provider && this._provider.getItems) {
+				const lazyLoadListItems = this._provider.getItems()[0]._getFlattenedListItems().lazyLoadListItems;
```

Comment:
Do you know how often this might get called? 

I'm always a little worried about the performance of these recursive methods. I did a bit of testing a year or two ago and found that for very large lists, the user experience degraded significantly, which is why the `d2l-list` component tends to make use of the `_selectionProvider` reference on the items. It still ends up being recursive, but it avoids the work that `getNestedList` does, and I'd found that was somewhat an improvement when working with large lists

Reply:
It would get called anytime selection info changes within a list. So every time selection changes we have to traverse the list tree to see if there are lazy load items.

Reply:
I think that's what I found to be janky with huge, deeply nested lists, as a result of all the logic to roll-up the selection to parents, etc. We might need to spend some time looking at ways to optimize this. Not sure it needs to be done in this PR thoug.

Reply:
Yeah, doing this in a later PR would be ideal. It's already getting pretty large as-is.